src/Repeater/DefaultRepeaterServer.ts

```diff
@@ -0,0 +1,173 @@
+        'Please make sure that repeater established a connection with host.'
+      );
+    }
+
+    this.socket.emit('deploy', {
+      repeaterId
+    });
+
+    const [result]: RepeaterServerDeployedEvent[] = await once(
+      this.socket,
+      'deployed'
+    );
+
+    return result;
+  }
+
+  public connect(hostname: string) {
+    this.socket = io(this.options.uri, {
+      parser,
+      path: '/api/ws/v1',
+      transports: ['websocket'],
+      reconnectionAttempts: this.DEFAULT_RECONNECT_TIMES,
+      auth: {
+        token: this.options.token,
+        domain: hostname
+      }
+    });
+
+    this.socket.on('connect_error', (error: Error) => {
+      logger.debug(`Unable to connect to the %s host`, this.options.uri, error);
+    });
+
+    this.createPingTimer();
+
+    logger.debug('Event bus connected to %s', this.options.uri);
+  }
+
+  public on(
+    event: RepeaterServerEvents,
+    handler: RepeaterServerEventHandlers
+  ): void {
+    if (event === 'reconnection_failed') {
+      this.onReconnectionFailed(
+        handler as RepeaterServerEventHandler<RepeaterServerReconnectionFailedEvent>
+      );
+    } else {
+      this.socket.on(event, (payload, callback) => {
+        this.processEventHandler(event, payload, (p) => handler(p), callback);
+      });
+    }
+  }
+
+  private onReconnectionFailed<
+    H extends RepeaterServerEventHandler<RepeaterServerReconnectionFailedEvent>
+  >(handler: H) {
+    this.socket.io.on('reconnect', () => {
+      this.latestReconnectionError = undefined;
+    });
+
+    this.socket.io.on('reconnect_error', (error) => {
+      this.latestReconnectionError = error;
+    });
+
+    this.socket.io.on('reconnect_attempt', (attempt) => {
+      logger.warn(
+        'Failed to connect (attempt %d/%d)',
+        attempt,
+        this.DEFAULT_RECONNECT_TIMES
+      );
+    });
+
+    this.socket.io.on('reconnect_failed', () => {
+      this.processEventHandler(
+        'reconnection_failed',
+        {
+          error: this.latestReconnectionError
+        },
+        handler
+      );
+    });
+  }
+
+  private processEventHandler<P>(
+    event: string,
+    payload: P,
+    handler: (payload: P) => unknown,
+    callback?: unknown
+  ) {
+    Promise.resolve(handler(payload))
+      .then((response) => {
+        if (typeof callback !== 'function') {
+          return;
+        }
+
+        callback(response);
+      })
+      .catch((error) => {
+        logger.debug(
+          'Error processing event "%s" with the following payload: %s. Details: %s',
+          event,
+          payload,
+          error
+        );
+        logger.error('Error: %s', error.message);
+      });
+  }
+
+  private createPingTimer() {
+    if (this.timer) {
+      clearInterval(this.timer);
+    }
+
+    this.timer = setInterval(
+      () => this.socket.volatile.emit('ping'),
+      10000
+    ).unref();
+  }
+
+  private clearPingTimer() {
+    if (this.timer) {
+      clearInterval(this.timer);
+    }
```

Comment:
The `clearInterval` accepts the undefined as well
```suggestion
    clearInterval(this.timer);
```

Reply:
According to `@types/node` and [Node.js docs](https://nodejs.org/dist/latest-v18.x/docs/api/timers.html#cleartimeouttimeout) it doesn't.

I see that we have `strictNullChecks` disabled, so all `undefined` ignored, but I would follow typing convention

<img width="634" alt="image" src="https://github.com/NeuraLegion/bright-cli/assets/9965996/af182ba5-df47-478f-8069-873e014f437c">

Reply:
In the Node.js documentation, it is clearly stated that Node.js accepts primitives as arguments, even if this is not explicitly covered in typings. Moreover,  Node.js versions < 18  >18, accept `undefined`  according to the typings, and even more, by the fact:
```ts
> clearTimeout(undefined)
undefined
```
So, seems there is a bug in typings for Node.js =>18 <19

Reply:
Actually, I realised that we have installed `"@types/node": "^10.17.59"`, so that's why we lack `undefined` in `clearInterval` typing